Improve project description (#345)
open-telemetry/opentelemetry-dotnet-instrumentation: docs/README.md

[![Slack](https://img.shields.io/badge/slack-@cncf/otel--dotnet--auto--instr-brightgreen.svg?logo=slack)](https://cloud-native.slack.com/archives/C01NR1YLSE7)

> This project aims to [add OpenTelemetry Instrumentation](https://opentelemetry.io/docs/concepts/instrumenting/#automatic-instrumentation)

Reviewer: ```suggestion
This project aims to add [OpenTelemetry Instrumentation](https://opentelemetry.io/docs/concepts/instrumenting/#automatic-instrumentation)
```

to .NET applications without requiring any changes to their source code.
To do that, it must:

1. Inject and setup the [OpenTelemetry .NET SDK](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/README.md#opentelemetry-net-sdk) into the application;
2. Add [OpenTelmetry Instrumentation](https://opentelemetry.io/docs/concepts/instrumenting/) to key packages and APIs used by the application;

> Moreover, if a package or API doesn't provide the necessary hooks
> to create a corresponding .NET instrumentation package,
> this project offers the capability to inject instrumentations during the application runtime,
> aka [monkey-patching](https://en.wikipedia.org/wiki/Monkey_patch) instrumentation.
> 
> See the [DESIGN.md](DESIGN.md) for an architectural overview of the project.

Reviewer 2: ```suggestion
The auto-instrumentation is capable of injecting instrumentations at runtime, 
a technique known as [monkey-patching](https://en.wikipedia.org/wiki/Monkey_patch). 
This allows to instrument specific packages or APIs that don't provide the necessary hooks
to generate .NET instrumentation packages.

See [DESIGN.md](DESIGN.md) for an architectural overview of the project.
```

## Status

> This project is in the early stages of development, starting with an initial seeding of code from the [.NET Tracer for Datadog APM](https://github.com/DataDog/dd-trace-dotnet). A [project board](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/projects/1) shows the current work in progress and the backlog.

Reviewer 2: ```suggestion
This project is in the early stages of development. A [project board](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/projects/1) shows the current work in progress and the backlog.
```

Author: I agree that the text gets better (shorter and clear). However, I prefer to ensure that @zacharycmontoya and @macrogreg are aware and agree with it before taking this change since, although Datadog is of course listed on the license etc, it removes a visible mention of the company name.

Author: We can discuss this later, will merge without this change.

Reviewer 3: Hey Paulo, we don't need to be visibly recognized here. You can remove this in another iteration
